File: js/test/Exchange/test.fetchLeverageTiers.js
'use strict';


const assert = require ('assert');
const testLeverageTier = require ('./test.leverageTier.js');

async function testFetchLeverageTiers (exchange, symbol) {
    const method = 'fetchLeverageTiers';
    const tiers = await exchange[method] (symbol);
    // const format = {
    //     'RAY/USDT': [
    //       {},
    //     ],
    // };
    assert (typeof tiers === 'object', exchange.id + ' ' + method + ' ' + symbol + ' must return an object. ' + exchange.json(tiers));
    const tierKeys = Object.keys (tiers);
    const arrayLength = tierKeys.length;
    assert (arrayLength >= 1, exchange.id + ' ' + method + ' ' + symbol + ' must have at least one entry. ' + exchange.json(tiers));
    console.log (exchange.id, method, 'fetched', arrayLength, 'entries, asserting each ...');
    for (let i = 0; i < arrayLength; i++) {
        const tiersForSymbol = tiers[ tierKeys[i] ];
        const arrayLengthSymbol = tiersForSymbol.length;
        assert (arrayLengthSymbol >= 1, exchange.id + ' ' + method + ' ' + symbol + ' must have at least one entry. ' + exchange.json(tiers));
        for (let j = 0; j < tiersForSymbol.length; j++) {
            testLeverageTier (exchange, method, tiersForSymbol[j]);
        }
    }
}

module.exports = testFetchLeverageTiers;

File: julia/Ccxt/test/exchange/test.fetchLeverageTiers.jl
using Test
using Ccxt
function testFetchLeverageTiers(exchange, skippedProperties, symbol)

    method = "fetchLeverageTiers";
    tiers = Base.fetch(fetchLeverageTiers(exchange, ["symbol"]));
    @test functions.ccxtruthy(isDictionary(exchange, tiers))
    tierKeys = objectKeys(tiers);
    assertNonEmtpyArray(testSharedMethods, exchange, skippedProperties, method, tierKeys, symbol);
    i = 0
    while functions.ccxtruthy(functions.ccxt_lt(i, length(tierKeys)))
        tiersForSymbol = get(tiers, Symbol(get(tierKeys, i + 1, nothing)), nothing);
        assertNonEmtpyArray(testSharedMethods, exchange, skippedProperties, method, tiersForSymbol, symbol);
        j = 0
        while functions.ccxtruthy(functions.ccxt_lt(j, length(tiersForSymbol)))
            testLeverageTier(exchange, skippedProperties, method, get(tiersForSymbol, j + 1, nothing));
            j += 1
        end
        i += 1
    end
    return true
end
